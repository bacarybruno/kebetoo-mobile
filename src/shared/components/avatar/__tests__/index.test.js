import { Image } from 'react-native'

import setupTest from 'Kebetoo/src/config/jest-setup'

import Avatar, { TextAvatar } from '../index'

const givenAvatar = setupTest(Avatar)({
  size: 20,
  fontSize: 20,
})

it('renders an Avatar with text', () => {
  const { wrapper } = givenAvatar({
    text: 'Bruno Bodian',
  })
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('renders an Avatar with image', () => {
  const { wrapper } = givenAvatar({
    src: 'https://avatars1.githubusercontent.com/u/14147533',
  })
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('displays TextAvatar if no image source is provided', () => {
  const { wrapper, props } = givenAvatar({
    text: 'Bruno Bodian',
    src: undefined,
  })
  expect(wrapper.root.findByType(TextAvatar).props.text).toBe(props.text)
})

it('displays ImageAvatar if an image source is provided', () => {
  const { wrapper, props } = givenAvatar({
    text: 'Bruno Bodian',
    src: 'https://avatars1.githubusercontent.com/u/14147533',
  })
  const firstLetter = props.text[0]
  const imageAvatar = wrapper.root.findByType(Image)
  expect(imageAvatar.props.source.uri).toBe(props.src)
  expect(wrapper.root.findAllByProps({ text: firstLetter }).length).toBe(0)
})
