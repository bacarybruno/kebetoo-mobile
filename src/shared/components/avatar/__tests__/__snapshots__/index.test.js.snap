// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an Avatar with image 1`] = `
<View
  style={
    Array [
      Object {
        "height": 30,
        "width": 30,
      },
      Object {
        "height": 20,
        "width": 20,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://avatars1.githubusercontent.com/u/14147533",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#007AFF",
          "borderRadius": 15,
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "borderRadius": 10,
        },
      ]
    }
    testID="image-avatar-https://avatars1.githubusercontent.com/u/14147533"
  />
</View>
`;

exports[`renders an Avatar with text 1`] = `
<View
  style={
    Array [
      Object {
        "height": 30,
        "width": 30,
      },
      Object {
        "height": 20,
        "width": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#007AFF",
          "borderRadius": 15,
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "borderRadius": 10,
        },
        Object {
          "backgroundColor": "#4dd0e1",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          false,
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "600",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          },
          false,
          false,
          Object {
            "backgroundColor": "transparent",
            "fontFamily": "System",
            "fontWeight": "600",
          },
          undefined,
          undefined,
          undefined,
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      B
    </Text>
  </View>
</View>
`;
